src/ui/public/notify/notifier.js

```diff
@@ -39,22 +34,61 @@ function closeNotif(cb, key) {
     // this === notif
     var i = notifs.indexOf(this);
     if (i !== -1) notifs.splice(i, 1);
-    if (this.timerId) this.timerId = clearTO(this.timerId);
+    if (this.timerId) {
+      Notifier.config.clearInterval(this.timerId);
+      this.timerId = undefined;
+    }
     if (typeof cb === 'function') cb(key);
   };
 }
 
+function cancelNotifTimer(cb, key) {
+  return function () {
+    if (this.timerId) {
+      Notifier.config.clearInterval(this.timerId);
+      this.timerId = undefined;
+    }
+
+    if (typeof cb === 'function') cb(key);
```

Comment:
To clarify: for public functions, I'd actually say there should be some assertion made about the type of `cb` early in the function so that it'd fail loudly if it were being misused, but we can be less cautious for private functions.